packages/numeric-input/src/NumericInput.js

```diff
@@ -0,0 +1,178 @@
+    /**
+     * Called after user changes the value of the field
+     */
+    onChange: PropTypes.func,
+    /**
+     * Called when user puts focus onto the field
+     */
+    onFocus: PropTypes.func,
+    /**
+     * Called as user changes the value of the field
+     */
+    onInput: PropTypes.func,
+    /**
+     * How much you want the arrows to move up or down
+     */
+    placeholder: PropTypes.string,
+    /**
+     * Initial text to display in the input
+     */
+    step: PropTypes.number,
+    /**
+     * Adds custom/overriding styles
+     */
+    stylesheet: PropTypes.func,
+    /**
+     * Starting value for the input
+     */
+    value: PropTypes.number,
+    /**
+     * The visual variant of the numeric input
+     */
+    variant: PropTypes.oneOf(variantTypes),
+    onClick: PropTypes.func,
+    disabled: PropTypes.bool,
+    defaultValue: PropTypes.number
+  };
+
+  static defaultProps = {
+    step: 1,
+    variant: "line"
+  };
+
+  render() {
+    const {
+      disabled,
+      onBlur,
+      onChange,
+      onFocus,
+      onInput,
+      placeholder,
+      step,
+      stylesheet,
+      value,
+      defaultValue: initialValue,
+      variant,
+      onClick,
+      ...otherProps
+    } = this.props;
+
+    const numericInputStylesheet = (styles, props, themeData) => {
+      const numericInputStyles = customStylesheet(styles, props, themeData);
+      return stylesheet
+        ? stylesheet(numericInputStyles, props, themeData)
+        : numericInputStyles;
+    };
+
+    return (
+      <div>
+        <ControlBehavior
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onMouseDown={this.props.onMouseDown}
+          onMouseEnter={this.props.onMouseEnter}
+          onMouseLeave={this.props.onMouseLeave}
+          onMouseUp={this.props.onMouseUp}
+        >
+          {({
+            hasFocus,
+            hasHover,
+            isPressed,
+            onBlur: handleBlur,
+            onFocus: handleFocus,
+            onMouseDown: handleMouseDown,
+            onMouseEnter: handleMouseEnter,
+            onMouseLeave: handleMouseLeave,
+            onMouseUp: handleMouseUp
+          }) => (
+            <SpinnerBehavior
+              onClick={onClick}
+              value={value}
+              onChange={onChange}
+              initialValue={initialValue}
+              step={step}
+              disabled={disabled}
+            >
+              {({
+                handleClick,
+                handleChange,
+                increment,
+                decrement,
+                value: controlledValue,
+                setValue
+              }) => {
+                const onDirectChange = event => {
+                  const newValue = event.target.value;
+                  setValue(newValue);
+                };
+
+                return (
+                  <div style={{ position: "relative" }}>
+                    <SpinnerPresenter
+                      value={Number(value)}
```

Comment:
Does `SpinnerPresenter` need most of these props? I did a quick browse and I don't think `hasFocus`, `hasHover`, `isPressed` is used on this level. Some of the event handlers may not need to be declared as well unless we are doing something specific with them.

Same with the `Input` on the props. I don't think we need a `className`, `disabled`, `placeholder`, `value`, some of the event handlers as you might be able to get them from `...props`.

If there is a reason for the props to be there, that's fine too.

Reply:
I took out the props for SpinnerPresenter, you were right I wasn't using most of them.
For Input, I took out className, but I had to leave the rest otherwise the behavior for the spinners and input wouldn't be tied together